replica/database.cc

```diff
@@ -1025,7 +1025,7 @@ future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_
     remove(*cf);
     cf->clear_views();
     co_await cf->await_pending_ops();
-    co_await _querier_cache.evict_all_for_table(cf->schema()->id());
+    co_await _querier_cache.evict_all_for_table(uuid);
     auto f = co_await coroutine::as_future(truncate(*cf, std::move(tsf), snapshot));
```

Comment:
The schema UUID is different than the UUID used to globally identify a ks.cf pair (which can be found through database::find_uuid()). I think it's worth adding a comment at least.

Reply:
And this is a very subtle point. We need to use strong types and only convert from plain uuid during communications.

Reply:
> That means this change is wrong, no?

yes. It's very easy to confuse table UUID and schema UUID, and using them in incorrect place means a bug, as they represent different things.

+1 for using strong types for identifiers.

e.g.:

replica::table::uuid_type

schema::uuid_type.

Reply:
But that's a change for another series.

Reply:
> The schema UUID is different than the UUID used to globally identify a ks.cf pair (which can be found through database::find_uuid()). I think it's worth adding a comment at least.

They are the same.
See https://github.com/scylladb/scylladb/blob/768a5c8b5a1eef0086eb9186924df7aee01d1b57/replica/database.cc#L959-L970

(Unlike the schema version() that returns the `table_schema_version` as UUID too)

Reply:
hmm schema->id() refers to the immutable table UUID, not schema version uuid. that's a bit confusing. perhaps id() should be renamed to table_id(). and switch to strong types for them as suggested by Avi.

Reply:
Will so as a followup

Reply:
#11207

Reply:
thanks!